splash/browser_tab.py

```diff
@@ -1116,6 +1185,140 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = self.get_id()
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class Event(object):
+    def __init__(self, storage, id, event):
+        self.storage = storage
+        self.id = id
+        self.event = event
+
+    def __getitem__(self, item):
+        return self.storage.get_event_property(self.id, item)
+
+    def preventDefault(self):
+        return self.storage.preventDefault.emit(self.id)
+
+    def stopImmediatePropagation(self):
+        return self.storage.stopImmediatePropagation.emit(self.id)
+
+    def stopPropagation(self):
+        return self.storage.stopPropagation.emit(self.id)
+
+    def remove(self):
+        return self.storage.remove_event(self.id)
+
+
+class EventHandlersStorage(QObject):
+    def __init__(self, parent, events_storage):
+        self.name = str(uuid.uuid1())
+        self.events_storage = events_storage
+        self.storage = {}
+        super(EventHandlersStorage, self).__init__(parent)
+
+    def add(self, func):
+        func_id = str(uuid.uuid1())
```

Comment:
calling `str(uuid.uuid1())` looks common - what do you think about extracting it to a function? it is also used in wait_for_resume implementation.